jobs/ContentShareJob.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace app\jobs;
+
+use app\models\Extension;
+use app\models\Wiki;
+use Yii;
+use Abraham\TwitterOAuth\TwitterOAuth;
+use app\models\ContentShare;
+use app\models\News;
+use yii\helpers\StringHelper;
+use yii\helpers\Url;
+use yii\queue\closure\Job;
+use yii\queue\Queue;
+
+class ContentShareJob extends Job
+{
+    /**
+     * @var int
+     */
+    public $objectTypeId;
+    /**
+     * @var int
+     */
+    public $objectId;
+    /**
+     * @var int
+     */
+    public $serviceId;
+
+    /**
+     * @param Queue $queue
+     */
+    public function execute($queue)
+    {
+        if (ContentShare::exists($this->objectTypeId, $this->objectId, $this->serviceId)) {
+            return;
+        }
+
+        $contentData = $this->getContentData();
+        if ($contentData === false) {
+            return;
+        }
+
+        $this->send($contentData['message'], $contentData['url']);
+    }
+
+    /**
+     * @return array|bool
+     */
+    protected function getContentData()
+    {
+        if ($this->objectTypeId == ContentShare::OBJECT_TYPE_NEWS) {
```

Comment:
This isn't flexible. How about https://github.com/yiisoft-contrib/yiiframework.com/pull/237/files#diff-5676ecadc2ffd32286d030d179d4f356R7?

Reply:
I agree that implementation through the extension of the base class is more convenient.

Reply:
We plan to connect other social networks, right? If yes, then I will add the opportunity to work with different social networks.